sim/src/mechanics/driving.rs

```diff
@@ -248,7 +248,25 @@ impl DrivingSimState {
         transit: &mut TransitSimState,
     ) -> bool {
         match car.state {
-            CarState::Crossing(_, _) => {
+            CarState::Crossing(time_int, dist_int) => {
+                let time_cross = now - time_int.start;
+                let avg_speed = Speed::from_dist_time(dist_int.length(), time_cross);
+                let route = car.router.head();
+                let max_speed = route.speed_limit(ctx.map).min(
+                    car.vehicle
+                        .max_speed
+                        .unwrap_or(Speed::meters_per_second(100.0)),
+                );
+                let speed_percent: u8 = ((avg_speed / max_speed) * 100.0) as u8;
```

Comment:
We should do this calculation in `analytics.rs` instead. Here, just fire an `Event::AgentCrossedLane(trip, lane, avg_speed, max_speed)` or something. The purpose of `Event` is to decouple the simulation details here from whatever choices we make later in `Analytics` about how to group/filter/process for data analysis/visualization.